### PR TITLE
feat: make csrf cookie security properties configurable

### PR DESCRIPTION
**Pull request by wgwz**
## Description

Closes: https://github.com/regen-network/regen-registry/issues/1596

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] targeted `dev` branch
- [x] provided a link to the relevant issue or specification
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed
- [ ] submit a PR against `master` branch after this one (and other PRs depending on this one, e.g. on regen-network/regen-web, if applicable) get merged

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**.*

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

File: server/middleware/csrf.ts [hunk 1]
```diff
@@ -1,17 +1,24 @@
...
+const secure = env.get('CSRF_COOKIE_SECURE').default('true').asBoolStrict();
 
 const options: DoubleCsrfConfigOptions = {
   getSecret: () => secret,
   cookieName: cookieName,
   cookieOptions: {
-    httpOnly: true,
-    sameSite: 'strict',
+    httpOnly: httpOnly,
+    sameSite: sameSite,
```
Comment (blushi) [inline server/middleware/csrf.ts:19]:
```suggestion
    httpOnly,
    sameSite,
```

File: server/middleware/csrf.ts [hunk 2]
```diff
@@ -1,17 +1,24 @@
...
     path: '/',
-    secure: true,
+    secure: secure,
```
Comment (blushi) [inline server/middleware/csrf.ts:21]:
```suggestion
    secure,
```

**Review by blushi — APPROVED** [review]
looks good
once deployed to master, we'll need to configure those env vars on heroku as well, right?
using "strict" for the csrf cookie and still "lax" for the session cookie?

**Comment by wgwz** [comment]
> looks good once deployed to master, we'll need to configure those env vars on heroku as well, right? using "strict" for the csrf cookie and still "lax" for the session cookie?

@blushi i think we might need to set both cookies samesite property to lax.
because if the csrf cookie is strict, while the session cookie is lax, it might lead to a weird situation.
where the user appears authenticated but the graphql requests return a csrf error.
i hadn't thought about that yet.
it's a little tricky to explain but the examples listed in [this SO answer][2] may help.

in any case, we would still maintain csrf protection even if we set the csrf cookie samesite to lax.
so using lax for both provides the desired UX and without sacrificing too much in terms of security (to the best of my knowledge).

i'm sort of making an assumption about the desired UX.
but basically it's that if a user is signed in to our application,
and they open a new tab, and somehow find a link back to our registry app,
and they click on it, they expect to still be signed in when they load the page.
if our user session cookie had samesite=strict,
they would not appear logged in, in this new tab.
i think some banks probably have this level of security,
but the common UX is that i'm still signed in.

just a side note, the csrf cookie name in prod should start with the `__HOST` prefix rather than the `__SECURE` prefix (as it is in staging).
this opts us in to some additional [cookie security features][1].
    
[1]: https://datatracker.ietf.org/doc/html/draft-west-cookie-prefixes-05#section-3.2
[2]: https://stackoverflow.com/a/59995877/4028706

**Pull request merged**